encoding/flag/flag.go

```diff
@@ -0,0 +1,170 @@
+			name = tag
+		}
+
+		// skip private variables and disabled flags
+		if tag == "-" || !field.CanSet() {
+			continue
+		}
+
+		if isAlias(field) {
+			if field.Type().String() == "time.Duration" {
+				d := field.Interface().(time.Duration)
+				f.values[name] = f.flagSet.String(name, d.String(), desc)
+				continue
+			}
+			if implementsMarshaler(field) {
+				b, _ := field.Interface().(encoding.TextMarshaler).MarshalText()
+				f.values[name] = f.flagSet.String(name, string(b), desc)
+				continue
+			}
+		}
+	switchStart:
+		switch field.Kind() {
+		// explicit list of unsupported types
+		case reflect.Array, reflect.Func, reflect.Chan, reflect.Complex64, reflect.Complex128, reflect.Interface, reflect.Map, reflect.Slice:
+			continue
+		case reflect.Int:
+			f.values[name] = f.flagSet.Int(name, field.Interface().(int), desc)
+		case reflect.Int8:
+			f.values[name] = f.flagSet.Int(name, int(field.Interface().(int8)), desc)
+		case reflect.Int16:
+			f.values[name] = f.flagSet.Int(name, int(field.Interface().(int16)), desc)
+		case reflect.Int32:
+			f.values[name] = f.flagSet.Int(name, int(field.Interface().(int32)), desc)
+		case reflect.Int64:
+			f.values[name] = f.flagSet.Int64(name, field.Interface().(int64), desc)
+		case reflect.Uint:
+			f.values[name] = f.flagSet.Uint(name, field.Interface().(uint), desc)
+		case reflect.Uint8:
+			f.values[name] = f.flagSet.Uint(name, uint(field.Interface().(uint8)), desc)
+		case reflect.Uint16:
+			f.values[name] = f.flagSet.Uint(name, uint(field.Interface().(uint16)), desc)
+		case reflect.Uint32:
+			f.values[name] = f.flagSet.Uint(name, uint(field.Interface().(uint32)), desc)
+		case reflect.Uint64:
+			f.values[name] = f.flagSet.Uint64(name, field.Interface().(uint64), desc)
+		case reflect.String:
+			f.values[name] = f.flagSet.String(name, field.Interface().(string), desc)
+		case reflect.Bool:
+			f.values[name] = f.flagSet.Bool(name, field.Interface().(bool), desc)
+		case reflect.Float32:
+			// note float32 has precision issues
+			f.values[name] = f.flagSet.Float64(name, float64(field.Interface().(float32)), desc)
+		case reflect.Float64:
+			f.values[name] = f.flagSet.Float64(name, field.Interface().(float64), desc)
+		case reflect.Ptr:
+			// todo dereference and handle. use recursion
+			field = field.Elem()
+			goto switchStart // easiest solution, but do we want a goto statement
+		case reflect.Struct:
+			// todo: Should we add a prefix for flags? structName-childVar or only
+			// support a struct if they implement a marshaler
+			if field.Type().String() == "time.Time" {
+				timeFmt := dField.Tag.Get(fmtTag)
+				timeFmt = getTimeFormat(timeFmt)
+				t := field.Interface().(time.Time)
+				f.values[name] = f.flagSet.String(name, t.Format(timeFmt), desc)
+				continue
+			}
+			if implementsMarshaler(field) {
+				b, _ := field.Interface().(encoding.TextMarshaler).MarshalText()
+				f.values[name] = f.flagSet.String(name, string(b), desc)
+			}
+		}
+	}
+	return f, nil
+}
+
```

Comment:
Should we move some helper methods into the encoding package?
isAlias, implementsUnmarshaler, implementsMarshaler and getTimeFormat

Reply:
Let's keep it organically grown like this for now. I think that we will at some point; it's just not clear yet what that will look like.